### PR TITLE
JP-1447: Fix set difference bug in get_object_info

### LINKED ISSUE #4932: calwebb_spec2 background step failing for NIS_WFSS exposure
_Issue [JP-1447](https://jira.stsci.edu/browse/JP-1447) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

The calwebb_spec2 pipeline is failing in the background step for a NIS_WFSS exposure when it tries to load the source catalog (to find background regions in the image) with the following traceback:
```java
2020-05-12 02:16:43,132 - stpipe.Spec2Pipeline.bkg_subtract - INFO - Extracting objects < abmag = 99.0
2020-05-12 02:16:43,132 - stpipe.Spec2Pipeline.bkg_subtract - INFO - Getting objects from /ifs/archive/test/jwst/store/doggett/tests/complete_long_05/dtest2/JWSTDP-2020.2.0-51~dc4bb2fc/2020-05-11-185809/sdp/asn_creation/cal/level2/jw00625-o025_t001_niriss_clear-f150w_cat.ecsv
2020-05-12 02:16:43,253 - stpipe.Spec2Pipeline.bkg_subtract - ERROR - Missing required columns in source catalog: {'nn_dist', 'aper70_abmag', 'aper50_vegamag', 'aper30_abmag_err', 'aper50_abmag', 'CI_30_70', 'aper_total_flux', 'isophotal_vegamag', 'sharpness', 'isophotal_vegamag_err', 'sky_orientation', 'aper50_abmag_err', 'is_star', 'aper_total_abmag_err', 'isophotal_flux', 'aper70_vegamag', 'aper30_flux_err', 'aper30_vegamag', 'aper_bkg_flux_err', 'aper_total_vegamag', 'semimajor_sigma', 'aper70_flux_err', 'aper50_vegamag_err', 'isophotal_area', 'aper_total_flux_err', 'ellipticity', 'aper70_flux', 'aper50_flux_err', 'aper30_vegamag_err', 'semiminor_sigma', 'aper70_vegamag_err', 'aper_bkg_flux', 'isophotal_abmag', 'aper_total_vegamag_err', 'isophotal_flux_err', 'aper30_abmag', 'aper50_flux', 'aper30_flux', 'aper_total_abmag', 'orientation', 'nn_abmag', 'isophotal_abmag_err', 'aper70_abmag_err', 'roundness'}
Traceback (most recent call last):
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/pipeline/calwebb_spec2.py", line 95, in process
    asn.filename
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/pipeline/calwebb_spec2.py", line 217, in process_exposure_product
    input = self.bkg_subtract(input, bkg_list)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/background/background_step.py", line 58, in process
    input_model, bkg_name, wlrange_name)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/background/background_sub.py", line 145, in subtract_wfss_bkg
    bkg_mask = mask_from_source_cat(input_model, wl_range_name)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/background/background_sub.py", line 229, in mask_from_source_cat
    grism_obj_list = create_grism_bbox(input_model, reference_files)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/assign_wcs/util.py", line 560, in create_grism_bbox
    skyobject_list = get_object_info(input_model.meta.source_catalog)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/assign_wcs/util.py", line 459, in get_object_info
    raise KeyError(err_text)
KeyError: "Missing required columns in source catalog: {'nn_dist', 'aper70_abmag', 'aper50_vegamag', 'aper30_abmag_err', 'aper50_abmag', 'CI_30_70', 'aper_total_flux', 'isophotal_vegamag', 'sharpness', 'isophotal_vegamag_err', 'sky_orientation', 'aper50_abmag_err', 'is_star', 'aper_total_abmag_err', 'isophotal_flux', 'aper70_vegamag', 'aper30_flux_err', 'aper30_vegamag', 'aper_bkg_flux_err', 'aper_total_vegamag', 'semimajor_sigma', 'aper70_flux_err', 'aper50_vegamag_err', 'isophotal_area', 'aper_total_flux_err', 'ellipticity', 'aper70_flux', 'aper50_flux_err', 'aper30_vegamag_err', 'semiminor_sigma', 'aper70_vegamag_err', 'aper_bkg_flux', 'isophotal_abmag', 'aper_total_vegamag_err', 'isophotal_flux_err', 'aper30_abmag', 'aper50_flux', 'aper30_flux', 'aper_total_abmag', 'orientation', 'nn_abmag', 'isophotal_abmag_err', 'aper70_abmag_err', 'roundness'}"
Traceback (most recent call last):
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/bin/strun", line 26, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 156, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/cmdline.py", line 346, in step_from_cmdline
    step.run(*positional)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/pipeline/calwebb_spec2.py", line 111, in process
    'One or more products failed to process. Failing calibration.'
RuntimeError: One or more products failed to process. Failing calibration.
----------------------------------------------------------------------
ERROR RUNNING STEP 'Spec2Pipeline':
    One or more products failed to process. Failing calibration.
----------------------------------------------------------------------
```
suggesting that the course catalog is missing some expected columns of data.

### PR DESCRIPTION
**Pull request by jwhite3**
`assign_wcs.util.get_object_info` was resulting in the incorrect reporting of missing columns in an input catalogue due to a set difference being in the wrong order. This has been fixed, and now the correct list of missing columns will be used in the error message

Resolves #4932 / [JP-1447](https://jira.stsci.edu/browse/JP-1447)

### CODE REVIEW

**Comment by hbushouse** [comment]
I think we might need other code changes to handle the new `source_catalog` column names where they're used in the WFSS source extraction code. Pinging Larry Bradley in JP-1447 for info about this.

**Comment by jwhite3** [comment]
Something to note: 

Another way of fixing this would be to just update `SkyObject` to include the updated column names, which I think would be better in the case of checking the "required" keys (which are held by `SkyObject`)

File: CHANGES.rst [hunk 1]
```diff
@@ -1,6 +1,10 @@
 0.16.1 (unreleased)
 ===================
 
+assign_wcs
+----------
+
+- Update get_object_info to check for and replace legacy keys for WFSS inputs [#4943]
```
Comment (hbushouse) [inline CHANGES.rst:7]: Not sure if it's necessary for sphinx, but you might need a blank line between lines 7 and 8.

File: jwst/assign_wcs/util.py [hunk 2]
```diff
@@ -473,8 +477,8 @@ def get_object_info(catalog_name=None):
                                  xcentroid=row['xcentroid'],
                                  ycentroid=row['ycentroid'],
                                  sky_centroid=row['sky_centroid'],
-                                 abmag=row['abmag'],
-                                 abmag_error=row['abmag_error'],
+                                 abmag=row['isophotal_abmag'],
```
Comment (hbushouse) [inline jwst/assign_wcs/util.py:480]: Ah good, so all we have to do is change the assignments here and then not worry about having to change the places where `abmag` and `abmag_error` are used in other parts of the code. 👍

**Comment by hbushouse** [comment]
> Something to note:
> 
> Another way of fixing this would be to just update `SkyObject` to include the updated column names, which I think would be better in the case of checking the "required" keys (which are held by `SkyObject`)

I assume that kind of change would then make it impossible for the code to work with "legacy" versions of the source catalog files? That's not necessarily a bad thing, just curious.

**Comment by jwhite3** [comment]
The checks on the keywords are actually happening for the keywords that are retrieved directly from `SkyObject` which is just a fancy `NamedTuple`, and so this wouldn't allow for backwards compatibility regardless (following the example of `sid`). 

If we wanted to allow for backwards compatibility, we could do that, but that would require a different approach than the current change and the suggested one above. I suppose we should find out if supporting the old versions of those keywords is necessary or if it _should_ raise an error

**Comment by hbushouse** [comment]
I don't see a pressing need for backwards compatibility. Was just curious if it was possible without too much work. We'll obviously need to update one of our regression test inputs, which runs the `calwebb_spec2` pipeline on NIRISS WFSS data. Instead of regenerating (from scratch) the source catalog for that test, it might be possible to simply edit the meta data for the column names?

**Comment by jwhite3** [comment]
The code changes are minimal (I already have a version with the updates to `SkyObject`). I should just be able to rename the columns and update the catalog file used with the regression tests now

**Pull request merged**